### PR TITLE
Added a maint mode check for admins to bypass maint page, replace wit…

### PR DESCRIPTION
**Pull request by gpickin**
…h maint banner.

https://ortussolutions.atlassian.net/browse/CONTENTBOX-740

image

### CODE REVIEW

**Comment by gpickin** [comment]
This is a nice solution, but for sites like Mapigator, that use the login service to login front end users, I was wondering if we want to make it only for admins, or a new permission for maint_mode_viewer.

What do you think?

**Comment by gpickin** [comment]
I think we should do the permission route, but then we'll need to add the permissions... how do we manage the DB updates for tickets like this @lmajano

File: modules/contentbox-ui/handlers/content.cfc [hunk 1]
```diff
@@ -29,8 +30,12 @@ component{
 	function preHandler( event, rc, prc ,action,eventArguments){
 		// Maintenance Mode?
 		if( prc.cbSettings.cb_site_maintenance ){
-			event.overrideEvent( "contentbox-ui:page.maintenance" );
-			return;
+			if( prc.oCurrentAuthor.getAuthorID() && prc.oCurrentAuthor.checkPermission( "MAINT_MODE_VIEWER" )  ){
```
Comment (lmajano) [inline modules/contentbox-ui/handlers/content.cfc]: Can you change it to a non-abbreviation approach:
`MAINTENANCE_MODE_VIEWER`

File: modules/contentbox-ui/handlers/content.cfc [hunk 2]
```diff
@@ -29,8 +30,12 @@ component{
...
+				html.addAsset( "#prc.cbRoot#/includes/js/maint.js" );
```
Comment (lmajano) [inline modules/contentbox-ui/handlers/content.cfc]: Can you rename the js file to `maintenance.js`

File: modules/contentbox-ui/handlers/content.cfc [hunk 3]
```diff
@@ -13,6 +13,7 @@ component{
 	property name="contentService"		inject="id:contentService@cb";
 	property name="commentService"		inject="id:commentService@cb";
 	property name="CBHelper"			inject="id:CBHelper@cb";
+	property name="html"				inject="HTMLHelper@coldbox";
```
Comment (lmajano) [inline modules/contentbox-ui/handlers/content.cfc]: No need to inject, part of supertype

File: modules/contentbox-ui/handlers/content.cfc (hunk 2 above)
Comment (lmajano) [inline modules/contentbox-ui/handlers/content.cfc]: Just change to `addAsset()`

**Comment by gpickin** [comment]
Updates made @lmajano

**Comment by gpickin** [comment]
I accidentally added the Perm to the updater directly to development :( 
Should I roll back dev, or not a big deal, this time?
Sorry.

**Comment by gpickin** [comment]
I reverted and added it to this branch.

**Pull request merged**